syncer/syncer.go

```diff
@@ -1132,580 +1127,632 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 		log.Debugf("[syncer] receive binlog event with header %+v", e.Header)
 		switch ev := e.Event.(type) {
 		case *replication.RotateEvent:
-			currentPos = mysql.Position{
-				Name: string(ev.NextLogName),
-				Pos:  uint32(ev.Position),
-			}
-			if currentPos.Name > lastPos.Name {
-				lastPos = currentPos
-			}
-
-			if shardingReSync != nil {
-				if currentPos.Compare(shardingReSync.currPos) == 1 {
-					shardingReSync.currPos = currentPos
-				}
-
-				if shardingReSync.currPos.Compare(shardingReSync.latestPos) >= 0 {
-					log.Infof("[syncer] sharding group %+v re-syncing completed", shardingReSync)
-					closeShardingSyncer()
-				} else {
-					log.Debugf("[syncer] rotate binlog to %v when re-syncing sharding group %+v", currentPos, shardingReSync)
-				}
-				continue
-			}
-			latestOp = rotate
-
-			if s.tracer.Enable() {
-				_, err := s.tracer.CollectSyncerBinlogEvent(traceSource, safeMode.Enable(), tryReSync, lastPos, currentPos, int32(e.Header.EventType), int32(latestOp))
-				if err != nil {
-					log.Infof("[syncer] tracer collect syncer binlog error: %v", errors.ErrorStack(err))
-				}
+			s.handleRotateEvent(
```

Comment:
can we define a struct to hold all parameters of these three event handler (handleRotateEvent, handleRowsEvent and handleQueryEvent), most of parameters of these three functions are same.

Reply:
agree! Rest also LGTM

Reply:
ping @kennytm